test/e2e/user_group_limit/user_group_limit_test.go

```diff
@@ -909,6 +923,190 @@ var _ = ginkgo.Describe("UserGroupLimit", func() {
+		// create a object of v1.Secret
+		secret := &v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: secretName,
+				Annotations: map[string]string{
+					"kubernetes.io/service-account.name": serviceAccountName,
+				},
+			},
+			Type: v1.SecretTypeServiceAccountToken,
+		}
+		time.Sleep(10 * time.Second)
+		_, err = kClient.CreateSecret(secret, namespace)
+		gomega.Ω(err).NotTo(HaveOccurred())
+		time.Sleep(10 * time.Second)
+		// Get the token value from the Secret
+		ginkgo.By("Getting the token value from the Secret...")
+		userTokenValue, err := kClient.GetSecretValue(namespace, secretName, "token")
+		gomega.Ω(err).NotTo(HaveOccurred())
+		// use deep copy not to hardcode the kubeconfig
+		config, err := kClient.GetKubeConfig()
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		// Modify the TLS configuration to remove cert/key files for token-based auth
+		newConf := rest.CopyConfig(config) // copy existing config
+		// Use token-based authentication instead of client certificates
+		newConf.CAFile = ""
+		newConf.CertFile = ""
+		newConf.KeyFile = ""
+		// Set the Kubernetes API server URL
+		newConf.BearerToken = userTokenValue
+		newConf.Username = ""
+		newConf.Password = ""
+		kubeconfigPath := filepath.Join(os.TempDir(), "test-user-config")
+		err = k8s.WriteConfigToFile(newConf, kubeconfigPath)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+		gomega.Ω(err).NotTo(HaveOccurred())
+		clientset, err = kubernetes.NewForConfig(config)
+		gomega.Ω(err).NotTo(HaveOccurred())
+		ginkgo.By("Creating Pod...")
+		pod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: podName,
+				Annotations: map[string]string{
+					"created-by": fmt.Sprintf("system:serviceaccount:%s:%s", namespace, serviceAccountName),
+					"user-token": userTokenValue, // Log the token in the annotation
+				},
+				Labels: map[string]string{"applicationId": "test-app"},
+			},
+			Spec: v1.PodSpec{
+				ServiceAccountName: serviceAccountName,
+				Containers: []v1.Container{
+					{
+						Name:  "nginx",
+						Image: "nginx",
+						Ports: []v1.ContainerPort{{ContainerPort: 80}},
+					},
+				},
+			},
+		}
+		_, err = clientset.CoreV1().Pods(namespace).Create(context.TODO(), pod, metav1.CreateOptions{})
+		gomega.Ω(err).NotTo(HaveOccurred())
+		createdPod, err := clientset.CoreV1().Pods(namespace).Get(context.TODO(), podName, metav1.GetOptions{})
+		gomega.Ω(err).NotTo(HaveOccurred())
+		ginkgo.By("Verifying User Info...")
+		userInfo, err := GetUserInfoFromPodAnnotation(createdPod)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		// user info should contain the substring "system:serviceaccount:default:test-user-sa"
+		gomega.Ω(strings.Contains(fmt.Sprintf("%v", userInfo), "system:serviceaccount:default:test-user-sa")).To(gomega.BeTrue())
+		// Remove the new kubeconfig file
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		os.Unsetenv("KUBECONFIG")
```

Comment:
I don't think we need to unset this variable, correct?

Reply:
Removed